packages/datetime/src/dateRangeInput.tsx

```diff
@@ -428,9 +662,12 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
 
     private getInputDisplayString = (boundary: DateRangeBoundary) => {
         const { values } = this.getStateKeysAndValuesForBoundary(boundary);
-        const { isInputFocused, inputString, selectedValue } = values;
+        const { isInputFocused, inputString, selectedValue, hoverString } = values;
 
-        if (isInputFocused) {
+        if (hoverString != null && !this.isControlled()) {
```

Comment:
I think that showing the hovered-range values in the fields is fine since it's an intermediate state - very similar to how controlled text inputs also allow you type freely inside. What you **can't** do however, is actually set those values `onBlur` `onEnter` etc. - that's the point in my mind where draw the distinction for consumer code to kick in.